### PR TITLE
Bump clap to 4.2.7

### PR DESCRIPTION
**Pull request by tan-yue**
The PR contains a reimplementation of the CLI commands using the new "derive" syntax.

The new syntax allows us to centralize a set of common arguments (snapfaas/src/cli.rs) and easily declares argument relationships. For example, one should specify exactly one of the tikv store and lmdb store (as this is required, the static DBENV is no longer needed and removed).

### CODE REVIEW

**Comment by alevy** [comment]
@tan-yue can you summarize the changes high level in the PR description? I get that this is upgrading the clap version, and taking advantage, it seems, of some new syntax to, e.g., centralize some of the common CLI arguments (worth noting). Does this _change_ any CLI arguments, or do all commands stay exactly the same for now, it's just an implementation difference?

**Comment by tan-yue** [comment]
> @tan-yue can you summarize the changes high level in the PR description? I get that this is upgrading the clap version, and taking advantage, it seems, of some new syntax to, e.g., centralize some of the common CLI arguments (worth noting). Does this _change_ any CLI arguments, or do all commands stay exactly the same for now, it's just an implementation difference?

The change is a reimplementation of the CLI commands using the new "derive" syntax.

The new syntax allows us to centralize a set of common arguments (snapfaas/src/cli.rs) and easily declares argument relationships. For example, one should specify exactly one of the tikv store and lmdb store (as this is required, the static DBENV is no longer needed and removed).
